src/compas_ags/ags/graphstatics.py

```diff
@@ -28,23 +28,25 @@
 from compas_ags.diagrams import ForceDiagram
 
 from compas_ags.ags.core import update_q_from_qind
-from compas_ags.ags.core import update_form_from_force
+from compas_ags.ags.core import update_primal_from_dual
 from compas_ags.ags.core import get_jacobian_and_residual
-from compas_ags.ags.core import parallelise_edges
 from compas_ags.ags.core import compute_jacobian
+from compas_ags.ags.core import parallelise_edges
+
+from compas_ags.utilities import check_equilibrium
 
 from compas_ags.exceptions import SolutionError
 
 
 __all__ = [
     'form_identify_dof',
     'form_count_dof',
-    'form_compute_nullspace',
     'form_update_q_from_qind',
     'form_update_from_force',
     'form_update_from_force_newton',
     'force_update_from_form',
     'force_update_from_constraints',
+    'update_diagrams_from_constraints',
 
     'form_update_q_from_qind_proxy',
     'form_update_from_force_proxy',
```

Comment:
we should remove these proxies but that can be done in a next round

Reply:
ok. I will open an issue with your comment